cherrybomb-engine/src/lib.rs

```diff
@@ -24,27 +24,43 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
 pub async fn run(config: &Config) -> anyhow::Result<Value> {
     verbose_print(config, None, "Starting Cherrybomb...");
 
-    // Reading OAS file to string
     verbose_print(config, None, "Opening OAS file...");
-    let oas_file = match std::fs::read_to_string(&config.file) {
-        Ok(file) => file,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error reading OAS file: {}", e));
-        }
-    };
-
-    // Parsing OAS file to JSON
-    verbose_print(config, None, "Parsing OAS file...");
-    let oas_json: Value = match serde_json::from_str(&oas_file) {
-        Ok(json) => json,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error parsing OAS file: {}", e));
+    // Reading OAS file to string let oas_json: Value;
+
+    let oas_json: Value;
+    let oas: OAS3_1;
+    match config.file.extension() {
+        Some(ext) => {
+            verbose_print(config, None, "Reading OAS file...");
+            let oas_file = match std::fs::read_to_string(&config.file) {
+                Ok(file) => file,
+                Err(e) => return Err(anyhow::anyhow!("Error opening OAS file: {}", e))?,
+            };
+            match ext.to_str() {
+                Some("json") => {
+                    verbose_print(config, None, "Parsing OAS file...");
+                    oas_json = match serde_json::from_str(&oas_file) {
+                        Ok(json) => json,
+                        Err(e) => return Err(anyhow::anyhow!("Error parsing OAS file: {}", e))?,
+                    };
+                }
+                Some("yaml") => {
+                    verbose_print(config, None, "Parsing OAS file...");
+                    oas_json = match serde_yaml::from_str(&oas_file) {
```

Comment:
Are you sure that serde_yaml::Value and serde_json::Value are the same for the compiler, I ran into trouble with this in the past, but I think there is a converter between them or something

Reply:
What do you mean the same for the compiler?  I used serde_yaml for the yaml struct and serde_json for the json struct.
They are completely separate types and have different definition.